### PR TITLE
feat: keeper l2

### PR DESCRIPTION
**Pull request by ashitakah**

### CODE REVIEW

File: solidity/test/integration/Ethereum Mainnet/AutomationVault.t.sol
Comment (wei3erHase) [inline solidity/test/integration/Ethereum Mainnet/AutomationVault.t.sol:1]: can we change the dir to `integration/mainnet` and `integration/sidechain`?

File: solidity/test/unit/Keep3rBondedRelayL2.t.sol [hunk 1]
```diff
@@ -0,0 +1,312 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {Test} from 'forge-std/Test.sol';
+
+import {IKeep3rJobWorkableRated} from '../../interfaces/external/IKeep3rJobWorkableRated.sol';
+
+import {
+  Keep3rBondedRelayL2,
+  IKeep3rRelayL2,
```
Comment (wei3erHase) [inline solidity/test/unit/Keep3rBondedRelayL2.t.sol]: unused

**Review by wei3erHase — DISMISSED** [review]
nit comment, the rest 👍

File: solidity/test/integration/Ethereum Mainnet/AutomationVault.t.sol
Comment (ashitakah) [reply solidity/test/integration/Ethereum Mainnet/AutomationVault.t.sol:1]: 👍

File: solidity/test/unit/Keep3rBondedRelayL2.t.sol (hunk 1 above)
Comment (ashitakah) [reply solidity/test/unit/Keep3rBondedRelayL2.t.sol]: 👍

**Review by wei3erHase — DISMISSED** [review]
🚀

**Review by 0xmoebius — APPROVED** [review]
lgtm

**Pull request merged**